### ISSUE TITLE
CURL options and Twitter "transfer closed with outstanding read data remaining"

### ISSUE THREAD

**Issue by kashum** [themattharris/tmhOAuth#150]
It would be nice to be able to set the HTTP version that curl uses, as a tmhOAuth configuration parameter, that defaults to 1.1, but can be changed to 1.0 by the caller.

About 4 months ago there was an issue with the Twitter API prematurely closing connections, and CURL generating a "transfer closed with outstanding read data remaining" error 18. Twitter subsequently fixed that, but the problem seems to have returned in the last few days.

Not sure if this is my fault or Twitter's, but the workaround for me was switch from HTTP/1.1 to HTTP/1.0, although there are other options you can try if you scan the dev.twitter forums.

In order to do this, I had to change tmhOAuth->curlit() to add CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_0 in the curl_setopt_array() call.

Which leads to this request, it would be nice to be able to have the HTTP version as a tmhOAuth configuration parameter, that defaults to 1.1, but can be changed to 1.0 by the caller.

**Comment by themattharris** [comment]
can you share the code you used that led to your getting "transfer closed with outstanding read data remaining". i'm concerned that you received that and would like to know what led to it (rather than work around it)

**Comment by kashum** [comment]
I'm not sure it's going to help, because Twitter seems to have fixed the issue again. But this is the code we are using:

``` PHP
$tmhOAuthConfig = array(
    'consumer_key' => 'blah',
    'consumer_secret' => 'blah',
    'token' => 'blah-blah',
    'secret' => 'blah'
);
$parameters = array(
    'count' => 3,
    'page' => 1,
    'screen_name' => 'ourscreenname'
);
$tmhOAuth = new tmhOAuth($tmhOAuthConfig);
$result = $tmhOAuth->user_request(array(
        'method'=>'GET',
        'url'=>$tmhOAuth->url('1.1/statuses/user_timeline'),
        'params'=>$parameters)
);

var_dump($result,$tmhOAuth->response);
```

Except the var_dump obviously.

**Comment by themattharris** [comment]
this turned out to be an issue on Twitter's end (https://dev.twitter.com/discussions/20009) which has since been resolved.

i'll add a way to set this to the config in 0.8.3 though.

**Comment by themattharris** [comment]
shipped in 0.8.3
